substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2246,13 +2253,23 @@ pub mod pallet {
 			let member_account = T::Lookup::lookup(member_account)?;
 			let mut member =
 				PoolMembers::<T>::get(&member_account).ok_or(Error::<T>::PoolMemberNotFound)?;
-			let current_era = T::Staking::current_era();
+			let current_era = T::StakeAdapter::current_era();
 
 			let bonded_pool = BondedPool::<T>::get(member.pool_id)
 				.defensive_ok_or::<Error<T>>(DefensiveError::PoolNotFound.into())?;
 			let mut sub_pools =
 				SubPoolsStorage::<T>::get(member.pool_id).ok_or(Error::<T>::SubPoolsNotFound)?;
 
+			let slash_weight = if bonded_pool.has_pending_slash() {
+				// apply slash if any before withdraw.
+				match Self::do_apply_slash(&member_account, None) {
+					Ok(_) => T::WeightInfo::apply_slash(),
+					Err(_) => T::WeightInfo::apply_slash_fail(),
+				}
```

Comment:
We should probably exit this function if there is a slash to apply, but we failed to do so. The `do_apply_slash` call has underlying effects in `pallet-delegated-staking`. Right now we're ignoring the error itself but keeping track of weight.

Reply:
If it fails it means the member has no slash to be applied right? Or do you see a codepath where there is slash to apply but fails and continues to withdraw?

Reply:
As discussed, let's use the `apply_slash_fail` when we get a `NothingToSlash` error and exit the function on any other error, as that would be bad state.